File: StructuralPatterns/Bridge/TypeScript/VectorRenderer.ts
import { Renderer } from './Renderer';

export class VectorRenderer implements Renderer {
    renderCircle(radius: number): void {
        console.log(`Drawing a circle of radius ${radius} with vector renderer.`);
    }
    renderSquare(size: number): void {
        console.log(`Drawing a square of size ${size} with vector renderer.`);
    }
}

File: StructuralPatterns/Bridge/CSharp/VectorRenderer.cs
using System;

namespace Bridge {
    public class VectorRenderer : IRenderer {
        public void RenderCircle(double radius) {
            Console.WriteLine($"Drawing a circle of radius {radius} with vector renderer.");
        }
        public void RenderSquare(double size) {
            Console.WriteLine($"Drawing a square of size {size} with vector renderer.");
        }
    }
}

